utils/kustomize.go

```diff
@@ -47,7 +53,7 @@ func WriteComponent(path string, resources []string) error {
 }
 
 // AddKustomizeComponent adds a component to an existing kustomization file.
-func AddKustomizeComponent(path, componentPath string) error {
+func AddKustomizeComponent(path string, componentPaths []string) error {
```

Comment:
Why did you convert `componentPath` to an array? it doesn't seem like you ever pass in more than one `string` ?

Reply:
Firstly I thought this would make it more scalable if we do. Otherwise we would have to sort our Kustomization every time a component was added which is significant overhead. Additionally this keeps its implementation similar to `AddKustomizeResource`. I am sure there was one case where we had multiple added components but I forget it now, let me see if I can find it and I'll come back to this.

Reply:
So you were right, there is no case where this is currently being used with array functionality. Should I change it back or keep it in case we every do something with the `opfcli` where we would need to add multiple components?

Reply:
`AddKustomizeResource` also doesn't look like it ever adds more than one item at a time, to me it makes more sense to just follow the format of the original `AddKustomizeComponent` and add one resource at a time in`AddKustomizeResource` because that's how you seem to be using it. Instead of having to constantly keep making new arrays just to add one item. Unless you actually have a situation where you are adding an array with more than one element, I think you can make this a single string element, same with `AddKustomizeResources`.

Reply:
There is a case where it adds more than one resource. In onboard you can create more than one namespace, and to deploy these changes to the cluster it calls `AddKustomizeResources` ([see code](https://github.com/operate-first/opfcli/pull/44/files#diff-b074f504f5bff6f4a2cd2f1607aa86248f0b5c64fd72ae87b3df866dbb9cfea0R80)) on the `cluster-scope/overlays/prod/<env>/<cluster>/kustomization.yaml` overlay (ie `prodOverlayPath`).

Reply:
you're right, I missed that part, okay keep it as is.